onnxruntime/core/framework/tensorprotoutils.cc

```diff
@@ -114,10 +114,114 @@ static Status UnpackTensorWithRawData(const void* raw_data, size_t raw_data_leng
       gsl::make_span(raw_data_bytes, raw_data_length), gsl::make_span(p_data, expected_size)));
   return Status::OK();
 }
+
+static Status GetExternalDataInfo(const ONNX_NAMESPACE::TensorProto& tensor_proto,
+                                  const ORTCHAR_T* tensor_proto_dir,
+                                  std::basic_string<ORTCHAR_T>& external_file_path,
+                                  onnxruntime::FileOffsetType& file_offset,
+                                  size_t& tensor_data_length) {
+  ORT_RETURN_IF(tensor_proto.data_location() != TensorProto_DataLocation_EXTERNAL,
+                "Tensor does not have external data to read from.");
+
+  ORT_RETURN_IF_NOT(
+      tensor_proto.data_type() != ONNX_NAMESPACE::TensorProto_DataType_UNDEFINED &&
+          tensor_proto.data_type() != ONNX_NAMESPACE::TensorProto_DataType_STRING,
+      "External data type cannot be UNDEFINED or STRING.");
+
```

Comment:
Use HasExternalData?

Reply:
updated code to use HasExternalData on line 123... 
check for string data type needs to be explicit it is not done in HasExternalData